### ISSUE TITLE
False positive with check-tag-names using js/ts decorator in @example

### ISSUE THREAD

**Issue by BBosman** [gajus/eslint-plugin-jsdoc#395]
I have a Java-/TypeScript decorator (named `transient`) inside my `@example` and the `check-tag-names` flags it as "Invalid JSDoc tag name". I would have expected everything between the backticks to be ignored by that rule?

Sample code:
````ts
/**
 * Registers the `target` class as a transient dependency; each time the dependency is resolved a new instance will be created.
 *
 * @param target - The class / constructor function to register as transient.
 *
 * @example ```ts
@transient()
class Foo { }
```
 */
export function transient<T>(target?: T): T {
  // ...
}
````

Rules: 
```json
{
  "jsdoc/check-tag-names": ["error", {
      "definedTags": [
        "chainable",
        "internal"
      ]
  }]
}
```

Am I doing something wrong, or is this an issue with the rule?
